btcutil/bech32/bech32.go

```diff
@@ -266,6 +266,15 @@ func DecodeNoLimit(bech string) (string, []byte, error) {
 	return hrp, data, err
 }
 
+// DecodeGeneric is identical to the existing DecodeNoLimit method, but will also
+// return the bech32 version that matches the decoded checksum.
+//
+// Note that the returned data is 5-bit (base32) encoded and the human-readable
+// part will be lowercase.
+func DecodeNoLimitGeneric(bech string) (string, []byte, Version, error) {
+	return decodeNoLimit(bech)
```

Comment:
How about we just export the `decodeNoLimit` and re-name it to `DecodeNoLimitWithVersion`? "Generic" doesn't really say that much IMO (at least in this context).

Reply:
I may have misunderstood your comment, are you suggesting renaming the private method decodeNoLimit  as a Public method named DecodeNoLimitWithVersion  and changing references to method in the package?

Reply:
Done!

Reply:
I have cleaned this up as requested.